docs/cep/windows/window-types/sliding-external-time.md

```diff
@@ -0,0 +1,34 @@
+---
+title: WINDOW SLIDING_EXTERNAL_TIME()
+---
+
+A sliding time window based on external time. It holds events that arrived during the last window `time` period from the external timestamp, and gets updated on every monotonically increasing timestamp.
+
+## Syntax
+
+    WINDOW SLIDING_EXTERNAL_TIME(<LONG> timestamp, <INT|LONG|TIME> time)
+
+## Query Parameters
+
+| Name        | Description            | Default Value | Possible Data Types | Optional | Dynamic |
+|-------------|------------------------|---------------|---------------------|----------|---------|
+| timestamp   | The time which the window determines as current time and will act upon. The value of this parameter should be monotonically increasing. |               | LONG     | No       | Yes     |
+| time | The sliding time period for which the window should hold events.      |          | INT LONG TIME    | No      | No      |
+
+## Example
+
+```sql
+CREATE STREAM StockEventStream (symbol string, price float, volume int);
```

Comment:
@koshy-thomas and @StoyanTc, this stream worker is invalid. I think the `eventTime` wants to be replaced with something, but I'm not sure what.

Reply:
@diana-macrometa 
for eventTime I think we can take it from input of stream..

CREATE STREAM cseEventStream (symbol string, price float, volume int, eventTime long);
CREATE SINK outputStream WITH (type='stream', stream='outputStream') (symbol string, price float);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int, eventTime long) SLIDING_EXTERNAL_TIME(eventTime, 20 sec) OUTPUT expired events;
@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;
@info(name = 'query1')
INSERT expired events INTO outputStream
SELECT symbol, price
FROM cseEventWindow;

But we also probably need to add a note that the eventTime attribute's value needs be a long value (reflecting the Unix timestamp in milliseconds)

Reply:
Can you give me an example of the long value that I can insert to test it?

Reply:
Actually, I found a long value converter, but the validator says this is invalid. Something about the attribute type.

```
CREATE STREAM cseEventStream (symbol string, price float, volume int, eventTime long);
CREATE SINK outputStream WITH (type='stream', stream='outputStream') (symbol string, price float);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int, eventTime long) SLIDING_EXTERNAL_TIME(1676060729, 20 sec) OUTPUT expired events;

@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;

@info(name = 'query1')
INSERT expired events INTO outputStream
SELECT symbol, price
FROM cseEventWindow;
```

Reply:
You can go to keep the stream open on one tab, keep the message sending open on one tab and use the value from here https://currentmillis.com/

Reply:
on the left hand side you can see the current unix time which is getting updated every millis

Reply:
Okay, that entering that value gives me an internal server error when I run this.

```
-- Metadata with name, description, and query language version.
@App:name("Temperature-Analytics")
@App:description("This stream worker creates a stream and query.")
@App:qlVersion("2")

CREATE STREAM cseEventStream (symbol string, price float, volume int, eventTime long);
CREATE SINK outputStream WITH (type='stream', stream='outputStream') (symbol string, price float);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int, eventTime long) SLIDING_EXTERNAL_TIME(1676061507472, 20 sec) OUTPUT expired events;

@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;

@info(name = 'query1')
INSERT expired events INTO outputStream
SELECT symbol, price
FROM cseEventWindow;
```

Reply:
It's not even a helpful error, just "Internal server error"

Reply:
@diana-macrometa - Max. int is 2^32 or 2,147,483,647, your value is 1,676,061,507,472. That's why you get `internal server error`.

Reply:
But all I did was copy the current time in milliseconds...

Reply:
Also, I ran it with the max value you gave me and got this error:

Error: There is no parameterOverload for 'TUMBLING_EXTERNAL_TIME' that matches attribute types '<INT, LONG>'. Supported parameter overloads are (<LONG> timestamp, <INT|LONG|TIME> window.time), (<LONG> timestamp, <INT|LONG|TIME> window.time, <INT|LONG|TIME> start.time), (<LONG> timestamp, <INT|LONG|TIME> window.time, <INT|LONG|TIME> start.time, <INT|LONG|TIME> timeout), (<LONG> timestamp, <INT|LONG|TIME> window.time, <INT|LONG|TIME> start.time, <INT|LONG|TIME> timeout, <BOOL> replace.with.batchtime).